support/scripts/install

```diff
@@ -3,67 +3,219 @@
-
-if [[ $(uname -s) == "Darwin" ]]; then
-    EXPECTED_SHA=$(grep -i "${os}" phoneinfoga_SHA256SUMS | awk '{print $1}')
-    DOWNLOAD_SHA=$(shasum -a 256 phoneinfoga_$os.tar.gz | awk '{print $1}')
-
-    if [[ "${EXPECTED_SHA}" == "${DOWNLOAD_SHA}" ]]; then
-        echo "Checksum verified"
-    else
-        echo "Error: Checksum validation failed."
+PHONEINFOGA_VERSION=$(curl -s https://api.github.com/repos/sundowndev/phoneinfoga/releases/latest | grep tag_name | cut -d '"' -f 4)
+
+AUTOMATIC=0
+SKIP_CHECKSUM=0
+
+usage() {
+
+    echo "PhoneInfoga Installer for version $PHONEINFOGA_VERSION"
+    echo
+    echo "DESCRIPTION: An installer script for downloading the latest release of PhoneInfoga. Without any arguments, $0 will detect your operating system and attempt to download the corresponding version of PhoneInfoga. Please submit an issue on GitHub if you encounter issues with this script."
+    echo
+    echo "USAGE: $0 [flag...] (-h|-s|-u)"
+    echo
+    echo "      -h | --help             Print this message and exit"
+    echo
+    echo "      -s | --skip-checksum    [Not Reccomended] Skip checksum validation"
+    echo "                              Only use this option if $0"
+    echo "                              is failing due to missing software."
+    echo
+    echo "      -u | --unsupported      Attempt to install on an unsupported OS."
+    echo "                              Useful when $0 is unable to detect"
+    echo "                              your OS automatically"
+    echo
+}
+
+
+validate_OS_type() {
+
+    if [[ -z "${OS}" ]]; then
+        echo "ERROR: Unable to determine your system type."
         exit 1
     fi
+}
+
+
+validate_supported_OS() {
+
+    SUPPORTED_OS_TYPES=(
+        "Linux_x86_64"
+        "Linux_armv6"
+        "Linux_arm64"
+        "Linux_i386"
+        "Darwin_x86_64"
+        "Darwin_arm64"
+    )
+
+    if [[ "${SUPPORTED_OS_TYPES[*]}" =~ "${OS}" ]]; then
+        return 0
+    else
+        echo "Error: $OS is not supported, installation will attempt to proceed manually."
+        echo
+        echo "Please check the releases page for a list of supported systems."
+        echo "https://github.com/sundowndev/phoneinfoga/releases"
+        echo
+        echo "Read more at https://sundowndev.github.io/phoneinfoga/install/"
+        echo
+        AUTOMATIC=1
+        return 1
+    fi
+}
+
+
+validate_checksum() {
+
+    local WGET_INSTALLED=false
+    which wget > /dev/null
+    [ $? -eq 1 ] || WGET_INSTALLED=true
+
+    echo "Validating checksum ..."
+    if $WGET_INSTALLED; then
+        wget --quiet --show-progress "https://github.com/sundowndev/phoneinfoga/releases/download/$PHONEINFOGA_VERSION/phoneinfoga_checksums.txt"
+    elif ! $WGET_INSTALLED; then
+        curl --progress-bar -LOC - "https://github.com/sundowndev/phoneinfoga/releases/download/$PHONEINFOGA_VERSION/phoneinfoga_checksums.txt"
+    fi
```

Comment:
It's more of a "just in case" thing, if for whatever reason some odd linux distro comes with wget installed instead of curl it won't fail.

Reply:
But it already uses curl everywhere else so I guess the script would fail earlier if curl wouldn't be installed

Reply:
curl is now the primary downloader, but I left in secondary checks for wget for any "just in case" scenarios. I think it's safer to have the option to fall back on it if there's ever a need.

Reply:
I get your point but I just don't want to maintain the wget dependency here, I'd rather keep this script simple and easy to maintain. If the user doesn't have curl, tar, and shasum/sha256sum, then the script cannot run.